File: TicTacToe/Private/TicTacToeActionsView.m
//
//  TicTacToeGridView.m
//  TicTacToe
//
//  Created by Hélio Mesquita on 23/01/20.
//  Copyright © 2020 Hélio Mesquita. All rights reserved.
//


#import "TicTacToeActionsView.h"

@interface TicTacToeActionsView ()

@end

@implementation TicTacToeActionsView

CGFloat touchAreaHeight = 80;

- (instancetype)init
{
    self = [super initWithFrame:UIScreen.mainScreen.bounds];
    if (self) {
        self.translatesAutoresizingMaskIntoConstraints = false;
    }
    return self;
}

-(void)configure {
    [self addSuperiorTapView];
    [self addInferiorTapView];
}

-(void) addSuperiorTapView {
    UIView * superiorView = [self createView:UIColor.clearColor];
    self.superiorView = superiorView;
    [NSLayoutConstraint activateConstraints:@[
         [superiorView.topAnchor constraintEqualToAnchor: self.topAnchor],
         [superiorView.trailingAnchor constraintEqualToAnchor: self.trailingAnchor],
         [superiorView.leadingAnchor constraintEqualToAnchor: self.leadingAnchor],
         [superiorView.heightAnchor constraintEqualToConstant:touchAreaHeight]
     ]
      ];
}

-(void) addInferiorTapView {
    UIView * inferiorView = [self createView:UIColor.clearColor];
    self.inferiorView = inferiorView;
    [NSLayoutConstraint activateConstraints:@[
        [inferiorView.bottomAnchor constraintEqualToAnchor: self.bottomAnchor],
        [inferiorView.trailingAnchor constraintEqualToAnchor: self.trailingAnchor],
        [inferiorView.leadingAnchor constraintEqualToAnchor: self.leadingAnchor],
        [inferiorView.heightAnchor constraintEqualToConstant:touchAreaHeight]
    ]
     ];
}

- (nonnull UIView *)createView:(nonnull UIColor *)color {
    UIView * touchView = [[UIView alloc]init];
    touchView.translatesAutoresizingMaskIntoConstraints = false;
    touchView.backgroundColor = color;
    [touchView setUserInteractionEnabled:true];
    [self addSubview: touchView];
    [self sendSubviewToBack: touchView];
    return  touchView;
}

@end

File: TicTacToeExample/TicTacToeActionsView.swift
//
//  ActionsView.swift
//  TicTacToeExample
//
//  Created by Hélio Mesquita on 21/01/20.
//  Copyright © 2020 Hélio Mesquita. All rights reserved.
//

import UIKit

class TicTacToeActionsView: UIView {

    let touchAreaHeight: CGFloat = 80
    var superiorView: UIView?
    var inferiorView: UIView?

    init() {
        super.init(frame: UIScreen.main.bounds)
        translatesAutoresizingMaskIntoConstraints = false
        configure()
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func configure() {
        addSuperiorTapView()
        addInferiorTapView()
    }

    func createView(_ color: UIColor) -> UIView {
        let baseline = UIView()
        baseline.translatesAutoresizingMaskIntoConstraints = false
        baseline.backgroundColor = color
        baseline.isUserInteractionEnabled = false
        self.addSubview(baseline)
        self.sendSubviewToBack(baseline)
        return baseline
    }

    func addSuperiorTapView() {
        let superiorView = createView(UIColor.clear)
        self.superiorView = superiorView
        NSLayoutConstraint.activate([
            superiorView.topAnchor.constraint(equalTo: topAnchor),
            superiorView.leadingAnchor.constraint(equalTo: leadingAnchor),
            superiorView.trailingAnchor.constraint(equalTo: trailingAnchor),
            superiorView.heightAnchor.constraint(equalToConstant: touchAreaHeight)
        ])
    }

    func addInferiorTapView() {
        let inferiorView = createView(UIColor.clear)
        self.inferiorView = inferiorView
        NSLayoutConstraint.activate([
            inferiorView.bottomAnchor.constraint(equalTo: bottomAnchor),
            inferiorView.leadingAnchor.constraint(equalTo: leadingAnchor),
            inferiorView.trailingAnchor.constraint(equalTo: trailingAnchor),
            inferiorView.heightAnchor.constraint(equalToConstant: touchAreaHeight)
        ])
    }
}


